### PR TITLE
Remove grpc client and project from providers

### PR DESCRIPTION
**Pull request by edwardrf**

### CODE REVIEW

File: src/pkg/cli/client/byoc/do/byoc.go [hunk 1]
```diff
@@ -461,10 +451,11 @@ func (b *ByocDo) Follow(ctx context.Context, req *defangv1.TailRequest) (client.
...
 func (b *ByocDo) TearDown(ctx context.Context) error {
-	_, err := b.BootstrapCommand(ctx, "down")
-	if err != nil {
-		return err
-	}
+	// FIXME: Do we need this? if so can we get it from remote?
+	// _, err := b.BootstrapCommand(ctx, "down")
+	// if err != nil {
+	// 	return err
+	// }
```
Comment (edwardrf) [inline src/pkg/cli/client/byoc/do/byoc.go:458]: Actually, is this behaviour consistent with the AWS teardown? aws provider only bring down the cd stack without removing the services afaik.

File: src/pkg/cli/client/byoc/aws/byoc.go [hunk 2]
```diff
@@ -51,9 +51,8 @@ var (
 type ByocAws struct {
 	*byoc.ByocBaseClient
 
-	cdImageTag      string
-	delegationSetId string
-	driver          *cfn.AwsEcs
+	cdImageTag string
```
Comment (lionello) [inline src/pkg/cli/client/byoc/aws/byoc.go:54]: let's remove this state

Comment (edwardrf) [reply src/pkg/cli/client/byoc/aws/byoc.go]: Done, also removed the lingering reference of it in DO

File: src/protos/io/defang/v1/fabric.proto [hunk 3]
```diff
@@ -529,6 +540,10 @@ message SubscribeResponse {
   ServiceState state = 4;
 }
 
+message GetServicesRequest {
+  string project = 2;
```
[the remark this answers is not in the document]
Comment (edwardrf) [reply src/protos/io/defang/v1/fabric.proto]: Updated to 1

File: src/pkg/cli/tail.go [hunk 4]
```diff
@@ -144,18 +145,21 @@ func (cerr *CancelError) Unwrap() error {
 	return cerr.error
 }
 
-func Tail(ctx context.Context, provider client.Provider, params TailOptions) error {
-	projectName, err := provider.LoadProjectName(ctx)
+func Tail(ctx context.Context, loader client.Loader, provider client.Provider, params TailOptions) error {
+	projectName, err := LoadProjectName(ctx, loader, provider)
```
[the remark this answers is not in the document]
Comment (edwardrf) [reply src/pkg/cli/tail.go:149]: Good point, using `params.Project`

**Pull request merged**